### PR TITLE
Replacement for #255

### PR DESCRIPTION
**Pull request by bertjwregeer**
Merging master into #255, and will add docs in this PR too.

### CODE REVIEW

File: docs/experimental/samesite.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Same-site Cookies
+=================
+
+The `Same-site cookie RFC
+<https://tools.ietf.org/html/draft-west-first-party-cookies-07>`_  updates
```
Comment (stevepiercy) [inline docs/experimental/samesite.txt]: Remove extra space:
```
07>`_  updates`
```

File: docs/experimental/samesite.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+`RFC6265 <https://tools.ietf.org/html/rfc6265>`_ to include a new cookie
+attribute named ``SameSite``.
+
+WebOb provides support for setting the ``SameSite`` attribute in it's cookie
```
Comment (stevepiercy) [inline docs/experimental/samesite.txt]: its

File: docs/experimental/samesite.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+API's, using the ``samesite`` keyword argument.
```
Comment (stevepiercy) [inline docs/experimental/samesite.txt]: APIs (neither possessive nor conjunction)

File: webob/cookies.py [hunk 4]
```diff
@@ -424,13 +434,44 @@ def _path_quote(v):
...
-    """ Generate a cookie value.  If ``value`` is None, generate a cookie value
-    with an expiration date in the past"""
+                secure=False, httponly=False, comment=None, samesite=None):
+    """ Generate a cookie value.
+
+    ``name``
+      The name of the cookie.
+
+    ``value``
+      The ``value`` of the cookie, if it is ``None`` it will generate a cookie
```
Comment (stevepiercy) [inline webob/cookies.py]:
```
The ``value`` of the cookie. If it is ``None``, it 
```

File: webob/cookies.py [hunk 5]
```diff
@@ -424,13 +434,44 @@ def _path_quote(v):
...
+      value with an expiration date in the past.
+
+    ``max_age``
+      The maximum age of the cookie used for sessioning (in seconds).
+      Default: ``None`` (browser scope).
+
+    ``path``
+      The path used for the session cookie. Default: ``'/'``.
```
Comment (stevepiercy) [inline webob/cookies.py]: Default: ``/``. (because we use a literal, no need to quote)

File: webob/cookies.py [hunk 6]
```diff
@@ -424,13 +434,44 @@ def _path_quote(v):
...
+
+    ``domain``
+      The domain used for the session cookie. Default: ``None`` (no domain).
+
+    ``secure``
+      The 'secure' flag of the session cookie. Default: ``False``.
+
+    ``httponly``
+      Hide the cookie from Javascript by setting the 'HttpOnly' flag of the
```
Comment (stevepiercy) [inline webob/cookies.py]: JavaScript (proper camelCasing)

Also perhaps ``HttpOnly``? Not sure on this one.

File: webob/cookies.py [hunk 7]
```diff
@@ -629,6 +672,10 @@ class CookieProfile(object):
       Hide the cookie from Javascript by setting the 'HttpOnly' flag of the
       session cookie. Default: ``False``.
 
+    ``samesite``
+      The 'SameSite' attribute of the cookie, can be either ``b"Strict"``,
```
Comment (stevepiercy) [inline webob/cookies.py]: Not sure on this one either. Perhaps ``SameSite``?

File: webob/cookies.py [hunk 8]
```diff
@@ -848,6 +910,10 @@ class SignedCookieProfile(CookieProfile):
...
```
Comment (stevepiercy) [inline webob/cookies.py]: Ditto: Perhaps ``SameSite``?

File: webob/response.py [hunk 9]
```diff
@@ -1001,6 +1002,12 @@ def set_cookie(self, name=None, value='', max_age=None,
            in the cookie, if it's ``False``, the ``HttpOnly`` flag will not
            be sent in the cookie.
 
+        ``samesite``
+
+          A string representing the ``SameSite`` attribute of the cookie or
+          ``None``. If samesite is ``None`` no ``SameSite`` value will be sent
```
Comment (stevepiercy) [inline webob/response.py]: Are the various casing and markups of samesite and SameSite intentional?

Also are there two contexts or one for samesite/SameSite? Each context should use consistent markup.

File: webob/response.py [hunk 10]
```diff
@@ -1001,6 +1002,12 @@ def set_cookie(self, name=None, value='', max_age=None,
...
+          in the cookie. Should only be ``b"Strict"`` or ``b"Lax"``
```
Comment (stevepiercy) [inline webob/response.py]: add period to end of sentence.

File: CHANGES.txt [hunk 11]
```diff
@@ -1,6 +1,14 @@
 Unreleased
 ----------
 
+Experimental Features
+~~~~~~~~~~~~~~~~~~~~~
+
+These features are experimental and may change at any point in the future.
+
+- The cookie API's now have the ability to set the SameSite attribute in both
```
Comment (stevepiercy) [inline CHANGES.txt]: APIs (this is neither possessive nor a conjunction)

Also see following comments about "SameSite" and "samesite".

File: CHANGES.txt [hunk 12]
```diff
@@ -1,6 +1,14 @@
...
+  webob.cookies.make_cookie and webob.cookies.CookieProfile.
```
Comment (stevepiercy) [inline CHANGES.txt]: Should these be marked up with reST?

File: webob/cookies.py (hunk 6 above)
Comment (bertjwregeer) [inline webob/cookies.py]: `httponly` is a kwarg, it is lowercase.

File: webob/cookies.py (hunk 7 above)
Comment (bertjwregeer) [inline webob/cookies.py]: Nope. Once again, kwarg.

File: webob/cookies.py (hunk 8 above)
Comment (bertjwregeer) [inline webob/cookies.py]: Nope.

File: webob/response.py (hunk 9 above)
Comment (bertjwregeer) [inline webob/response.py]: Yes. This is on purpose. One is a kwarg, the other is the attribute name in the RFC.

File: CHANGES.txt (hunk 12 above)
Comment (bertjwregeer) [inline CHANGES.txt]: No, this is in CHANGES.txt, I haven't marked any of the other things up either. I might add `` around them.

File: webob/cookies.py (hunk 6 above)
Comment (stevepiercy) [inline webob/cookies.py]: Right on the first one (the term), I meant the second (in the definition).

File: webob/response.py (hunk 9 above)
Comment (stevepiercy) [inline webob/response.py]: OK, then should it be this for consistency and indicate that it is a kwarg?
```
 ``samesite``
```

Comment (stevepiercy) [inline webob/response.py]: Also you switch between:

```
'SameSite'
```
and
```
``SameSite``
```

**Pull request merged**